Use .NET 6 code (#31123)
dotnet/docs: docs/azure/sdk/dependency-injection.md

# Dependency injection with the Azure SDK for .NET

> This article demonstrates how to register Azure service clients from the [latest Azure SDKs for .NET](https://azure.github.io/azure-sdk/releases/latest/index.html#net) for [dependency injection in an ASP.NET Core app](/aspnet/core/fundamentals/dependency-injection). Every ASP.NET Core app starts up by using the instructions provided in the `Program.cs` file.

Reviewer: ```suggestion
This article demonstrates how to register Azure service clients from the [latest Azure SDKs for .NET](https://azure.github.io/azure-sdk/releases/latest/index.html#net) for [dependency injection in an ASP.NET Core app](/aspnet/core/fundamentals/dependency-injection). Every ASP.NET Core app starts up by using the instructions provided in the _Program.cs_ file.
```

## Register client

> In the `Program.cs` file, register a client for each service:

Reviewer: ```suggestion
In the _Program.cs_ file, register a client for each service:
```

## Use the registered clients

> With the clients registered in the `Program.cs` file, you can now use them:

Reviewer: ```suggestion
With the clients registered in the `AddAzureClients`, you can now use them:
```

You can add any options from <xref:Azure.Core.ClientOptions> into the JSON file's `AzureDefaults` section. One of the options is the retry policy. For more information, see [Configure a new retry policy](#configure-a-new-retry-policy).

> Since the `Configuration` object is injected from the host you can use the following code in the `Program.cs` file:

Reviewer: ```suggestion
Since the `Configuration` object is a member of the `builder` instance, you can configure secrets:
```

```csharp
builder.Services.AddAzureClients(clientBuilder =>
{
    // Add a KeyVault client
    clientBuilder.AddSecretClient(builder.Configuration.GetSection("KeyVault"));

    // Add a storage account client
    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("Storage"));

    // Use DefaultAzureCredential by default
    clientBuilder.UseCredential(new DefaultAzureCredential());

    // Set up any default settings
    clientBuilder.ConfigureDefaults(builder.Configuration.GetSection("AzureDefaults"));
});

builder.Services.AddControllers();
```

## Configure multiple service clients with different names

> Assume you have two storage accounts: one for private information and one for public information. Your app transfers data from the public to private storage account after some operation. You need to have two storage service clients. To set up these clients in the `Program.cs` file:

Reviewer: ```suggestion
Assume you have two storage accounts: one for private information and one for public information. Your app transfers data from the public to private storage account after some operation. You need to have two storage service clients. To set up these clients in the _Program.cs_ file:
```
